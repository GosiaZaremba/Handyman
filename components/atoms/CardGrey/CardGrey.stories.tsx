import React from 'react';
import {storiesOf} from '@storybook/react-native';
import {CardGrey} from './CardGrey';
import {action} from '@storybook/addon-actions';
import {text} from '@storybook/addon-knobs';

storiesOf('atoms/CardGrey', module)
  .addDecorator(story => story())
  .add('default', () => <CardGrey children={''} />);
