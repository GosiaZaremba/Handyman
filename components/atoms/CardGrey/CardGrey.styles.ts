import {StyleSheet} from 'react-native';
import {Colors} from '../../../constants/colors';

export default StyleSheet.create({
  container: {
    backgroundColor: Colors.star.grey,
    borderRadius: 5,
    paddingHorizontal: 5,
    paddingVertical: 2,
    marginRight: 6,
  },
});
