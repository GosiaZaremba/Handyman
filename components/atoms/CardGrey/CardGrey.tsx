import React, {ReactNode} from 'react';
import {View} from 'react-native';
import styles from './CardGrey.styles';

export type Props = {
  children?: ReactNode;
  testID?: string;
};

export const CardGrey: React.FC<Props> = ({children, testID}) => {
  return (
    <View style={styles.container} testID={testID}>
      {children}
    </View>
  );
};
