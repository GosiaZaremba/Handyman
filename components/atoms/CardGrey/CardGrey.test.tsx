import React from 'react';
import {render} from '@testing-library/react-native';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import {CardGrey} from './CardGrey';
import {View} from 'react-native';

describe('Atom CardGrey', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <CardGrey testID="atom-cardGrey">
          <View />
        </CardGrey>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure children are rendered correctly', () => {
    const {getByTestId} = render(
      <CardGrey testID="atom-cardGrey">
        <View />
      </CardGrey>,
    );
    const cardGrey = getByTestId('atom-cardGrey');
    expect(cardGrey.props.children.type.displayName).toBe('View');
  });
});
