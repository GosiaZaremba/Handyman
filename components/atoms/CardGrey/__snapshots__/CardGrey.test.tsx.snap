// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom CardGrey Component Snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e1e1e1",
      "borderRadius": 2,
      "marginRight": 6,
      "paddingHorizontal": 5,
      "paddingVertical": 2,
    }
  }
  testID="atom-cardGrey"
>
  <View />
</View>
`;
